lms/djangoapps/instructor/views/api.py

```diff
@@ -2112,3 +2134,24 @@ def spoc_gradebook(request, course_id):
         'staff_access': True,
         'ordered_grades': sorted(course.grade_cutoffs.items(), key=lambda i: i[1], reverse=True),
     })
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_sales_admin
+@require_POST
+def make_invoice_transaction(request, course_id):
+    """
+    Adding invoice transaction  (payment or refund) for the Invoice.
+    """
+    course_key = CourseKey.from_string(course_id)
+    invoice_id = request.POST.get('invoice_id')
+    amount_type = request.POST.get('amount_type')
+    amount = int(request.POST.get('amount'))
+    comments = request.POST.get('comments')
+    if invoice_id and amount and amount_type:
+        if amount_type == 'refund':
+            amount *=  -1
```

Comment:
Please also quantize the amount (ensure that it has at most 2 decimal places).  I believe we use Python's `decimal` module to do this elsewhere in the shopping cart.